### PR TITLE
struct: Support reflect from struct with non-anonymous structure pointer

### PR DESCRIPTION
**Pull request by ygj6**
### What problem should be fixed?
Type reflection panic on embedded pointer field #225
### Have you added test cases to catch the problem?
Yes,here are some test cases.

### CODE REVIEW

File: struct.go [hunk 1]
```diff
@@ -595,7 +595,7 @@ func (s *Section) reflectFrom(val reflect.Value) error {
 			continue
 		}
 
-		if (tpField.Type.Kind() == reflect.Ptr && tpField.Anonymous) ||
+		if (tpField.Type.Kind() == reflect.Ptr && (tpField.Anonymous || (field.Elem().Kind() == reflect.Struct))) ||
 			(tpField.Type.Kind() == reflect.Struct && tpField.Type.Name() != "Time") {
```
[the remark this answers is not in the document]
Comment (ygj6) [reply struct.go:599]: It just adds the judgment of the structure pointer. I should write it like this
```go
if (tpField.Type.Kind() == reflect.Ptr && tpField.Anonymous) ||
   (tpField.Type.Kind() == reflect.Ptr && tpField.Type.Elem().Kind() == reflect.Struct) ||
```

Another question: Why should we add the judgment of anonymous pointers before? 
According to my understanding, the logic in this section is used to process nested structures, which is irrelevant to anonymity. If an anonymous non-structure pointer is transferred, panic:NumField of non-struct type may occur. Therefore, can (tpField.Type.Kind() == reflect.Ptr && tpField.Anonymous) be deleted?

If I don't get it right, please point out.😂

File: struct_test.go [hunk 2]
```diff
@@ -554,6 +554,31 @@ None        =
...
+				Enable  bool   `ini:"enable"`
+				Type    string `ini:"type"`
+				Address string `ini:"addr"`
+				Name    string `ini:"name"`
+			}
+			type Cfg struct {
+				Rpc *Rpc `ini:"rpc"`
+			}
+
+			So(cfg.ReflectFrom(&Cfg{&Rpc{true,"type","address","name"}}), ShouldBeNil)
```
[the remark this answers is not in the document]
Comment (ygj6) [reply struct_test.go:572]: I'll fix it later.

File: struct.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (unknwon) [reply struct.go]: >Therefore, can (tpField.Type.Kind() == reflect.Ptr && tpField.Anonymous) be deleted?

If tests don't fail, then I guess we can remove it :D

[the remark this answers is not in the document]
Comment (ygj6) [reply struct.go]: Yes,I have modified the PR, all existing test cases pass. 
Please review it again.

**Review by unknwon — APPROVED** [review]

**Comment by unknwon** [comment]
https://github.com/go-ini/ini/releases/tag/v1.60.2 has been tagged for this merge.

**Pull request merged**